src/mantarray_desktop_app/mc_simulator.py

```diff
@@ -111,6 +86,15 @@ def __init__(
         self._status_code_bits: bytes = bytes(0)
         self._reset_status_code_bits()
 
+    @property
+    def in_waiting(self) -> int:
```

Comment:
Is this always guaranteed to be accurate? What if `leftover_read_bytes` is `not None` but there are still things in the queue to be pulled out?

I'm not saying we need to guarantee its accuracy. But if it's not accurate, that should be clearly denoted in the docstring (with as much detail as possible about what it is guaranteed to do, and what situations would make it inaccurate).

The `in_waiting` for `pyserial` is guaranteed to be accurate right?